server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java

```diff
@@ -115,10 +130,12 @@ public ReadOnlyEngine(
         TranslogStats translogStats,
         boolean obtainLock,
         Function<DirectoryReader, DirectoryReader> readerWrapperFunction,
-        boolean requireCompleteHistory
+        boolean requireCompleteHistory,
+        boolean allowOldIndices
```

Comment:
Rather than overloading here with the allow flag, could you put this as extra into indexSettings?  Then you could tie it as a special setting for SS only indices when parsing settings?